doc/y_cable/design_doc.md

```diff
@@ -113,105 +143,119 @@ Vendors can have several implementations/ways to use this concept
 - Now for xcvrd to use this solution, all it has to do is instantiate the class objects defined by importing the appropriate module. This should happen when xcvrd starts or if there is a change event  (xcvrd inserted/removed)
 
-```
-	from credo import y_cable
-	y_cable_port_Ethernet0 = Ycable(port)
+```python
+        # import the vendor to module mapping
+        from sonic_y_cable import y_cable_vendor_mapping
+ 
+        # This vendor/part_name determination subroutines will either read through eeprom
+        # or else read from the TRANSCEIVER_INFO_TABLE inside state db
+
+        vendor_name = get_vendor_name(port)
+
+        part_name = get_part_name(port)
 
+        module = y_cable_vendor_mapping.mapping[vendor_name][part_name]
 
+        from sonic_y_cable import vendor_name.part_name.module.YCable as YCable
+
+        y_cable_port_Ethernet0 = YCable(port)
+
+        # and to use the Y cable APIS
+        try:
+            y_cable_port_Etherne0.toggle_mux_to_torA()
+        except:
+            helper_logger.log(not able to toggle_mux_to_torA)
 
-	# and to use the Y cable APIS
-	try:
-	    y_cable_port_etherne0.toggle_mux_to_torA()
-	except:
-	    helper_logger.log(not able to toggle_mux_to_torA)
 ```
 #### Rationale
 
   - xcvrd can maintain a dict of these objects an whenever it needs to call the API for the correct port (in this case physical port) it can easily achieve so by indexing into the instance of the physical port and calling the class object.
 
-```
-
-	Y_cable_ports_instances = {}
+```python
+        Y_cable_ports_instances = {}
 
-	# appending instances in the dictionary
-	Y_cable_ports_instances[physical_port] = Ycable(port)
+        # appending instances in the dictionary
+        Y_cable_ports_instances[physical_port] = YCable(port)
 
-	# and to use the Y cable APIS
-	try:
-	    Y_cable_ports_instances[physical_port].toggle_mux_to_torA()
-	except:
-	    helper_logger.log(not able to toggle_mux_to_torA)
+        # and to use the Y cable APIS
+        try:
+            Y_cable_ports_instances[physical_port].toggle_mux_to_torA()
+        except:
+            helper_logger.log(not able to toggle_mux_to_torA)
 ```
 
 
 ### How does cli interact with Y cable api's
-- Another thing that is required is once we do have a vendor to module mapping, we need to map appropriate port to a module as well
+- Another thing that is required is once we do have a a mapping from vendor/part number to the appropriate Y-cable module to load, we need to map appropriate port to a module as well
 
 #### Background
 
-  - Another requirement we have is Cli also requires to interact with the Ycable directly. This basically implies that all  Ycable vendor packages needs to be imported inside SONiC-utilities/host as well
+  - Another requirement we have is Cli also requires to interact with the YCable directly. This basically implies that all  Ycable vendor packages needs to be imported inside SONiC-utilities/host as well
```

Comment:
s/YCable/Y-Cable/

Reply:
fixed

Reply:
There were two instances; you missed the second one.

Reply:
fixed